### PR TITLE
Config validators refactoring

### PR DESCRIPTION
**Pull request by AnatolyPopov**
* Moving common validators into single place e.g. additional module
* Adding a couple of new validators
* Validators usage cleanup

### CODE REVIEW

File: commons/src/main/java/io/aiven/kafka/tiered/storage/config/validators/NonEmptyPassword.java [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package io.aiven.kafka.tieredstorage.storage.config;
+package io.aiven.kafka.tiered.storage.config.validators;
```
Comment (jeqo) [inline commons/src/main/java/io/aiven/kafka/tiered/storage/config/validators/NonEmptyPassword.java:17]: is the change from `tieredstorage` to `tiered.storage` on purpose?

Comment (AnatolyPopov) [reply commons/src/main/java/io/aiven/kafka/tiered/storage/config/validators/NonEmptyPassword.java:17]: Nope, fixed

**Review by jeqo — COMMENTED** [review]
Looking good! Left a few minor comments.

Could we also add some tests for the new validators?

File: storage/gcs/src/test/java/io/aiven/kafka/tieredstorage/storage/gcs/GcsStorageConfigTest.java [hunk 2]
```diff
@@ -49,6 +49,26 @@ void minimalConfig() {
         }
     }
 
+    @Test
+    void invalidEndpointUrl() {
```
[the remark this answers is not in the document]
Comment (AnatolyPopov) [reply storage/gcs/src/test/java/io/aiven/kafka/tieredstorage/storage/gcs/GcsStorageConfigTest.java:53]: This is controversial I think. Quite some people think that people should prefer camel case for abbreviations in method names(I'm not saying whether it's right or wrong), we also Have GcsStorage for example. But at the same time this one just corresponds to the naming enpoint url in other places in the code.

[the remark this answers is not in the document]
Comment (AnatolyPopov) [reply storage/gcs/src/test/java/io/aiven/kafka/tieredstorage/storage/gcs/GcsStorageConfigTest.java:53]: So I guess it should be also ValidUrl instead in corresponding validator

File: commons/build.gradle [hunk 3]
```diff
@@ -0,0 +1,19 @@
+plugins {
+    id 'java'
+}
+
+group = 'io.aiven.kafka.tiered.storage'
```
[the remark this answers is not in the document]
Comment (AnatolyPopov) [reply commons/build.gradle]: Fixed

**Review by jeqo — COMMENTED** [review]
Thanks for adding the tests! It looks good, just a final comment on the build file.

File: commons/build.gradle
[the remark this answers is not in the document]
Comment (AnatolyPopov) [reply commons/build.gradle:1]: Fixed

**Review by jeqo — APPROVED** [review]

**Pull request merged**